integration-cli/docker_cli_config_test.go

```diff
@@ -46,46 +47,36 @@ func (s *DockerSuite) TestConfigHttpHeader(c *check.C) {
 	}`
 
 	err = ioutil.WriteFile(tmpCfg, []byte(data), 0600)
-	if err != nil {
-		c.Fatalf("Err creating file(%s): %v", tmpCfg, err)
-	}
+	c.Assert(err, checker.IsNil)
 
 	cmd := exec.Command(dockerBinary, "-H="+server.URL[7:], "ps")
 	out, _, _ := runCommandWithOutput(cmd)
 
-	if headers["User-Agent"] == nil {
-		c.Fatalf("Missing User-Agent: %q\nout:%v", headers, out)
-	}
+	c.Assert(headers["User-Agent"], checker.NotNil, check.Commentf("Missing User-Agent"))
 
-	if headers["User-Agent"][0] != "Docker-Client/"+dockerversion.VERSION+" ("+runtime.GOOS+")" {
-		c.Fatalf("Badly formatted User-Agent: %q\nout:%v", headers, out)
-	}
+	c.Assert(headers["User-Agent"][0], checker.Equals, "Docker-Client/"+dockerversion.VERSION+" ("+runtime.GOOS+")", check.Commentf("Badly formatted User-Agent,out:%v", out))
+
+	c.Assert(headers["Myheader"], checker.NotNil)
+	c.Assert(headers["Myheader"][0], checker.Equals, "MyValue", check.Commentf("Missing/bad header,out:%v", out))
 
-	if headers["Myheader"] == nil || headers["Myheader"][0] != "MyValue" {
-		c.Fatalf("Missing/bad header: %q\nout:%v", headers, out)
-	}
 }
 
 func (s *DockerSuite) TestConfigDir(c *check.C) {
 	cDir, err := ioutil.TempDir("", "fake-home")
-	c.Assert(err, check.IsNil)
+	c.Assert(err, checker.IsNil)
 	defer os.RemoveAll(cDir)
 
 	// First make sure pointing to empty dir doesn't generate an error
-	out, rc := dockerCmd(c, "--config", cDir, "ps")
+	_, rc := dockerCmd(c, "--config", cDir, "ps")
```

Comment:
You can remove the assert and just use `dockerCmd`. The exit code (`rc` here) will never be different to 0 (because if it is it means there has been an error, and then `dockerCmd` will already have failed the test).

```go
dockerCmd(c, "--config", cDir, "ps")
```